webapp/src/Entity/ExecutableFile.php

```diff
@@ -138,4 +139,12 @@ public function isExecutable(): bool
     {
         return $this->isExecutable;
     }
+
+    /**
+     * @ORM\PreRemove()
+     */
+    public function disallowDelete(): void
+    {
+        throw new \RuntimeException('An executable file cannot be deleted');
```

Comment:
If you delete an immutable executable, can you still delete the file? Should you be able to even?

Reply:
You can't delete an immutable executable either.